File: core/src/de/project/ice/utils/StringJoiner.java
package de.project.ice.utils;

import java.util.Iterator;

public abstract class StringJoiner
{
    public static String join(Iterable<String> iterable, String delimiter)
    {
        Iterator<String> iterator = iterable.iterator();
        if (!iterator.hasNext())
        {
            return "";
        }
        StringBuilder sb = new StringBuilder(iterator.next());
        while (iterator.hasNext())
        {
            sb.append(delimiter);
            sb.append(iterator.next());
        }
        return sb.toString();
    }
}

File: core/src/de/project/ice/utils/StringJoiner.kt
package de.project.ice.utils

fun Iterable<Any>.join(delimiter: String): String {
    val iterator = this.iterator()
    if (!iterator.hasNext()) {
        return ""
    }
    val sb = java.lang.StringBuilder(iterator.next().toString())
    while (iterator.hasNext()) {
        sb.append(delimiter)
        sb.append(iterator.next().toString())
    }
    return sb.toString()
}